src/test/java/unicash/logic/parser/UniCashParserTest.java

```diff
@@ -149,6 +148,31 @@ public void parseCommand_summary() throws Exception {
                 CommandType.SUMMARY.getMainCommandWord() + " 3") instanceof SummaryCommand);
     }
 
+    @Test
+    public void parseCommand_findCommand() throws Exception {
+
+        List<String> keywords = Arrays.asList("n/foo", "l/bar", "c/baz");
+        List<String> parsedKeywords = Arrays.asList("foo", "bar", "baz");
+        List<Predicate<Transaction>> predicateList = new ArrayList<>();
+
+        TransactionContainsAllKeywordsPredicate allKeywordsPredicate =
+                new TransactionContainsAllKeywordsPredicate(predicateList);
+
+        allKeywordsPredicate.addNameKeyword((parsedKeywords.get(0)));
+        allKeywordsPredicate.addCategoryKeyword(parsedKeywords.get(2));
+        allKeywordsPredicate.addLocationKeyword(parsedKeywords.get(1));
+
+
+        FindCommand findCommand = (FindCommand) parser.parseCommand(
+                CommandType.FIND.getMainCommandWord() + " " + keywords
+                        .stream()
+                        .collect(Collectors
+                                .joining(" ")));
```

Comment:
This expression seems super complex, maybe best to split the joined keywords into another variable? Like

```java
var joinedKeywords = keywords.stream().collect(Collectors.joining(" "));
var findCommand = (FindCommand) parser.parseCommand(
    CommandType.FIND.getMainCommandWord() + " " + joinedKeywords
);
```

Reply:
Fixed